synapse/tests/test_lib_ast.py

```diff
@@ -4524,3 +4524,171 @@ async def test_ast_varlistset(self):
             text = '($x, $y) = (1)'
             with self.raises(s_exc.StormRuntimeError):
                 await core.nodes(text)
+
+    async def test_ast_functypes(self):
+
+        async with self.getTestCore() as core:
+            q = '''
+            function foo() {
+                for $n in { return((newp,)) } { $lib.print($n) }
+            }
+            [ it:dev:str=test ]
+            $foo()
+            '''
+            msgs = await core.stormlist(q)
+            self.stormNotInPrint('newp', msgs)
```

Comment:
Are there positive assertions we can make over these tests? Like a node was yielded or something specifically happened? Or that no errors occured?

I'm thinking of this specifically after the guid constructor tests failed to assert that no errors occurred at all and hide the fact that a `None` value got into the pipeline.

Reply:
we could add a `stormHasNoErr` check and verify we got a node in the output, do you think that is sufficient?

Reply:
:+1: That should work

Reply:
updated